pkg/providers/tinkerbell/create.go

```diff
@@ -62,6 +62,11 @@ func (p *Provider) PreCAPIInstallOnBootstrap(ctx context.Context, cluster *types
 }
 
 func (p *Provider) PostBootstrapSetup(ctx context.Context, clusterConfig *v1alpha1.Cluster, cluster *types.Cluster) error {
+	return p.AddHardwareInCluster(ctx, cluster)
+}
+
+// AddHardwareInCluster adds all the hardwares provided in the hardware.csv file to the cluster.
```

Comment:
I think we should avoid mentioning the hardware CSV in the comment given this code should be oblivious to the use of a hardware CSV.